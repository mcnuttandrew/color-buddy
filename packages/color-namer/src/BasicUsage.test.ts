import { test, expect } from "vitest";
import { nameColor } from "color-buddy-color-namer";
import { Color } from "color-buddy-palette";

test("Simple Usage", () => {
  [
    "#FF0000",
    "#00FF00",
    "#0000FF",
    "#FFFF00",
    "#00FFFF",
    "#FF00FF",
    "#FFFFFF",
    "#000000",
    "#efebe8",
    "#f5f1f0",
    "#f9f8f7",
    "#f0eae8",
    "#f5f0ef",
    "#faf5f4",
    "#f5eddc",
    "#faf4e7",
    "#fefaf1",
    "#e9ebe0",
    "#f0f1e8",
    "#f7f7f0",
    "#ebebeb",
    "#f2f2f2",
    "#f9f9f9",
    "#dfe7e8",
    "#e9f0f1",
    "#f2f9f9",
    "#f3ebf3",
    "#f7f1f6",
    "#fbf7fb",
    "#e2e6f0",
    "#e9edf5",
    "#f0f3fa",
    "#e6ecf0",
    "#eef1f3",
    "#f2f6f7",
    "#ddebf0",
    "#e7f1f5",
    "#f0f7fa",
    "#dfedeb",
    "#e9f3f2",
    "#f3faf9",
    "#e8edda",
    "#eff3e3",
    "#f7faf0",
    "#f5ead7",
    "#f6eee3",
    "#faf5f0",
    "#f9e9e0",
    "#f9eee8",
    "#f9f3ef",
  ].forEach((hex) => {
    const color = Color.colorFromString(hex);
    const name = nameColor(color, { numResults: 1 })[0];
    expect(name, `${color} named correctly`).toMatchSnapshot();
  });
});
