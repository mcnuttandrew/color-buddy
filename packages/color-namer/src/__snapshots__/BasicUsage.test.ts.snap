// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Usage 1`] = `"brightred"`;

exports[`Simple Usage 2`] = `"brightgreen"`;

exports[`Simple Usage 3`] = `"blue"`;

exports[`Simple Usage 4`] = `"yellow"`;

exports[`Simple Usage 5`] = `"cyan"`;

exports[`Simple Usage 6`] = `"neonpurple"`;

exports[`Simple Usage 7`] = `"white"`;

exports[`Simple Usage 8`] = `"black"`;

exports[`Simple Usage 9`] = `"white"`;

exports[`Simple Usage 10`] = `"white"`;

exports[`Simple Usage 11`] = `"white"`;

exports[`Simple Usage 12`] = `"white"`;

exports[`Simple Usage 13`] = `"white"`;

exports[`Simple Usage 14`] = `"white"`;

exports[`Simple Usage 15`] = `"offwhite"`;

exports[`Simple Usage 16`] = `"offwhite"`;

exports[`Simple Usage 17`] = `"white"`;

exports[`Simple Usage 18`] = `"lightgrey"`;

exports[`Simple Usage 19`] = `"lightgrey"`;

exports[`Simple Usage 20`] = `"white"`;

exports[`Simple Usage 21`] = `"white"`;

exports[`Simple Usage 22`] = `"white"`;

exports[`Simple Usage 23`] = `"white"`;

exports[`Simple Usage 24`] = `"lightgrey"`;

exports[`Simple Usage 25`] = `"white"`;

exports[`Simple Usage 26`] = `"white"`;

exports[`Simple Usage 27`] = `"white"`;

exports[`Simple Usage 28`] = `"white"`;

exports[`Simple Usage 29`] = `"white"`;

exports[`Simple Usage 30`] = `"white"`;

exports[`Simple Usage 31`] = `"white"`;

exports[`Simple Usage 32`] = `"white"`;

exports[`Simple Usage 33`] = `"white"`;

exports[`Simple Usage 34`] = `"white"`;

exports[`Simple Usage 35`] = `"white"`;

exports[`Simple Usage 36`] = `"paleblue"`;

exports[`Simple Usage 37`] = `"white"`;

exports[`Simple Usage 38`] = `"white"`;

exports[`Simple Usage 39`] = `"paleblue"`;

exports[`Simple Usage 40`] = `"white"`;

exports[`Simple Usage 41`] = `"white"`;

exports[`Simple Usage 42`] = `"offwhite"`;

exports[`Simple Usage 43`] = `"offwhite"`;

exports[`Simple Usage 44`] = `"offwhite"`;

exports[`Simple Usage 45`] = `"offwhite"`;

exports[`Simple Usage 46`] = `"white"`;

exports[`Simple Usage 47`] = `"white"`;

exports[`Simple Usage 48`] = `"white"`;

exports[`Simple Usage 49`] = `"white"`;

exports[`Simple Usage 50`] = `"white"`;
